Improved Cascading-Colors UX

mirisuzanne opened the pull request:
## Cute animal pic
_Because everyone likes pictures of [animals](https://unsplash.com/t/animals)._

![quizzical  bird](https://images.unsplash.com/photo-1593280449110-f482285dd97c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2547&q=80)

## Link to Trello card (if applicable)
_If this PR relates to a Trello card, don't forget to reference this PR on the card as well._

https://trello.com/c/twS6Ay2H/28-i-want-to-improve-the-view-color-options

## Description
_The commit messages say what you did; this should explain why and/or how._
- [x] Description is in Trello card



## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

Play with the color menu, and see how it works

## Show me
_Provide screenshots/animated gifs/videos if necessary._

![Screenshot 2020-06-30 17 50 07](https://user-images.githubusercontent.com/104161/86187918-4165c380-bafa-11ea-9313-4e51edaf0156.jpg)
![Screenshot 2020-06-30 17 50 26](https://user-images.githubusercontent.com/104161/86187920-4296f080-bafa-11ea-9f89-43eba4662812.jpg)
![Screenshot 2020-06-30 17 50 42](https://user-images.githubusercontent.com/104161/86187921-43c81d80-bafa-11ea-8de1-c692fc6520f4.jpg)

mirisuzanne commented:
![](https://github.trello.services/images/mini-trello-icon.png) [I want to improve the view/color options](https://trello.com/c/twS6Ay2H/28-i-want-to-improve-the-view-color-options)

stacyk commented on content/_includes/forms.macros.njk:
```diff
@@ -99,6 +99,96 @@ params:
+{# @docs
+label: fieldset
+category: Fields
+note: Create a
```
Create a what? Don't leave me hanging 🧗‍♂️

stacyk commented on src/scss/forms/_radios.scss:
```diff
@@ -0,0 +1,49 @@
+@use '../config';
+
+/// # Radio Buttons
+/// @group input-radio
+
+/// @group input-radio
```
duplicate lines

mirisuzanne replied:
These are actually two different blocks. The first is adding an h1 to the input-radio page, and the second is attached to the code block below. They could maybe use more detail, but they aren't duplicate.

stacyk replied:
I sorta wondered if it was there for just the heading... ok, makes sense.

stacyk commented on src/scss/patterns/_dropdown.scss:
```diff
@@ -21,11 +21,11 @@
+  border: thin solid var(--active);
   box-shadow: 0 0 var(--quarter-shim) var(--shadow);
-  max-width: 14em;
+  max-width: calc(100vw - var(--page-margin) * 2);
```
curious how we can up with this equation?

mirisuzanne replied:
the viewport-width, minus the two (left and right) page margins, should be the full content width.

stacyk replied:
ah, so that is for small screens. Got it.

mirisuzanne commented:
@stacyk I do know there are a lot of people who prefer low-contrast options for accessibility. It's best to provide a default that's high-contrast for the most common a11y need, but I think allowing that to be changed is also an a11y feature. I tried to limit the contrast bar to a reasonable range, while still allowing it to go low-contrast if desired.

mirisuzanne commented:
![](https://github.trello.services/images/mini-trello-icon.png) [I want the light/dark menu to close on blur](https://trello.com/c/FGb5nJbu/88-i-want-the-light-dark-menu-to-close-on-blur)

mirisuzanne commented:
While addressing @stacyk's review, I also implemented a CSS-only click-anywhere-to-close trick that I saw github using on their menus. If that looks good, I think I've addressed all the comments here.

jgerigmeyer commented on src/scss/config/_focus.scss:
```diff
@@ -0,0 +1,12 @@
+/// is applied by default on element-focus,
+/// but some patterns (like radio-buttons) also require
+/// focus-ring applied to a sibling element.
+/// @group config-focus
```
@mirisuzanne This is missing a group-mapping in `.sassdocrc`.

jgerigmeyer replied:
@mirisuzanne Given your comment on the SVGs, I think this is the only remaining thing before merge?

jgerigmeyer commented on src/scss/patterns/_dropdown.scss:
```diff
@@ -13,6 +13,26 @@
   position: relative;
 }
 
+// Dropdown Button
+// ---------------
+/// Generates a fixed overlay of the viewport when open.
+/// The overlay is part of the summary button,
+/// between the dropdown itself and the rest of the page,
+/// and acts as a click target to close the dropdown.
+/// @group dropdown
+[data-btn~='summary'] {
+  [open] > & {
+    @include tools.z-index('dropdown');
+
+    @include tools.before('') {
+      @include tools.position(0 fixed);
+      @include tools.z-index('bump');
+      cursor: default;
+      display: block;
+    }
+  }
+}
+
```
😍

jgerigmeyer commented on content/_includes/icons/audio.svg:
```diff
@@ -3,8 +3,6 @@
   xmlns="http://www.w3.org/2000/svg"
   aria-hidden="true"
   focusable="false"
-  width="20px"
-  height="20px"
```
@mirisuzanne So. Herman runs these through SVGO, which by default removes the `viewBox` attr if it matches the height/width exactly. Then we run those optimized SVGs through gulp-svg-symbols, which breaks (with our custom sprite-sheet, which relies on `viewBox`) if `viewBox` is missing...

I'm not clear on the tradeoffs here, or when we do or don't need height/width. Probably this is a bug that should be fixed in Herman -- let's discuss.

mirisuzanne replied:
well, they don't do exactly the same thing - but they do overlap a lot.

- `viewBox` is better for most things. We don't ever want to remove it. if we're choosing, it should absolutely be what we keep.
- I like _also_ having height/width, because they provide a default size for the icon. It's less essential, but it means they'll remain reasonably-sized until we explicitly resize them in CSS (which is simple when we want it).

You can see the difference here: https://cdpn.io/mirisuzanne/debug/002175ac77b34d172f1cdf769673c899

So… I'm ok with this, and it shouldn't hold up the  branch, but I don't love it as a restriction. Long term, I'm not sure we should have Herman do so much "optimization" as a documentation tool.

jgerigmeyer replied:
Yeah, that makes sense to me. It'd be a quick fix to Herman to stop optimizing SVGs entirely...

mirisuzanne commented:
@jgerigmeyer every little fix becomes a bigger fix. I think this also addresses https://trello.com/c/XTKW4RCc/87-heman-sizes-not-showing-preview-correctly-maybe-only-when-in-a-css-var now…

mirisuzanne commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Heman sizes not showing preview correctly (maybe only when in a CSS Var?)](https://trello.com/c/XTKW4RCc/87-heman-sizes-not-showing-preview-correctly-maybe-only-when-in-a-css-var)
